File: DSAlgorithms/Leetcode/arrays/arrays_hash/179-largest_num.cpp
class Solution {
public:
    static bool compare(std::string a, std::string b)
    {
        return a + b > b + a;
    }
    string largestNumber(vector<int>& nums) {
        std::vector<std::string> vec;
        std::string result = "";
        for(int i = 0; i < nums.size(); i++)
            vec.push_back(std::to_string(nums[i]));
        std::sort(vec.begin(), vec.end(), compare);
       if(vec[0] == "0")
            return "0";
        for(auto val: vec)
           result += val; 
        
        return result;
    }
};

File: dsa_algo/leetcode/ts/src/arrays_hash/179-largest_num.ts
const  largestNumber = (nums: number[]): string =>{
    let largest = nums
        .map((n) => n.toString())
        .sort((x: any, y:any) => y + x - (x + y))
        .join('');

    return largest[0] === '0' ? '0' : largest;
};